mmv1/third_party/terraform/website/docs/r/cloudbuild_worker_pool.html.markdown

```diff
@@ -0,0 +1,120 @@
+
+---
+subcategory: "Cloud Build"
+layout: "google"
+page_title: "Google: google_cloudbuild_worker_pool"
+sidebar_current: "docs-google-cloudbuild-worker-pool"
+description: |-
+  Configuration for custom WorkerPool to run builds
+---
+
+# google\_cloudbuild\_worker\_pool
+
+Definition of custom Cloud Build WorkerPools for running jobs with custom networking.
+
+-> This resource is not currently public, and requires allow-listing of projects prior to use.
+
+~> **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
+See [Provider Versions](https://terraform.io/docs/providers/google/guides/provider_versions.html) for more details on beta resources.
+
+
+## Example Usage
+
+```hcl
+resource "google_cloudbuild_worker_pool" "pool" {
+  name = "my-pool"
+  location = "europe-west1"
+  worker_config {
+    disk_size_gb = 100
+    machine_type = "e2-standard-4"
+    no_external_ip = false
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `location` -
+  (Required)
+  The location for the resource
+  
+* `name` -
+  (Required)
+  User-defined name of the `WorkerPool`.
+  
+
+- - -
+
+* `network_config` -
+  (Optional)
+  Network configuration for the `WorkerPool`.
+  
+* `project` -
+  (Optional)
+  The project for the resource
+  
+* `worker_config` -
+  (Optional)
+  Configuration to be used for a creating workers in the `WorkerPool`.
```

Comment:
```suggestion
  Configuration for workers in the `WorkerPool`.
```

Reply:
I think the current wording is actually more correct here. Also this comes from the API directly

I don't exactly know how workers are created, but I believe the worker pool contains the config that is used to set up the worker somewhat on-demand.